pkg/splunk/enterprise/upgrade.go

```diff
@@ -0,0 +1,315 @@
+package enterprise
+
+import (
+	"context"
+	"fmt"
+
+	enterpriseApi "github.com/splunk/splunk-operator/api/v4"
+	splclient "github.com/splunk/splunk-operator/pkg/splunk/client"
+	splcommon "github.com/splunk/splunk-operator/pkg/splunk/common"
+	appsv1 "k8s.io/api/apps/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	rclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// helps in mock function
+var GetClusterInfoCall = func(ctx context.Context, mgr *indexerClusterPodManager, mockCall bool) (*splclient.ClusterInfo, error) {
```

Comment:
Should we move it to the testing package?

Reply:
but his is to mock. its better to be here so code reading becomes easy.